### ISSUE TITLE
Avoid flagging `require-param` with @implements present (and possibly also @augments/@extends)

### ISSUE THREAD

**Issue by brettz9** [gajus/eslint-plugin-jsdoc#100]
For the `TransparentColor` function in this example:

```js
/**
 * @interface
 * @param {string} name The color name
 * @returns {object} A color object
 */
function Color (name) {}

/**
 * Function representing a transparent color.
 * @implements {Color}
 */
function TransparentColor (a) {}
```

...eslint-plugin-jsdoc's `jsdoc/require-param` rule complains with:

> Missing JSDoc @param "a" declaration

However, the [JSDoc (for @implements)](http://usejsdoc.org/tags-implements.html) states:

> If you do not document one of the symbols in the implementation, JSDoc will automatically use the interface's documentation for that symbol.

This would seem to suggest that with `@implements` present, it is expected that values need only be added if they differ from the interface they are implementing.

While JSDoc even states:

> You do not need to add the @implements tag to each member of the implementation (for example, the implementation's instance methods).

...it would be helpful if eslint-plugin-jsdoc could at least avoid the error when `@implements` is explicit on the block. In SVGEdit, we have a number of these real-world uses which are getting unduly flagged, and it would be a maintenance head-ache to have to duplicate parameter and return information on implementing objects as well as on the interface.

([\@augments](http://usejsdoc.org/tags-augments.html) (or its synonym `@extends`) should probably behave this way as well. I see you already support this optionally with [@override](http://usejsdoc.org/tags-override.html).)

----

Btw, I also filed an issue against jsdoctypeparser (at https://github.com/Kuniwak/jsdoctypeparser/issues/47 ) to allow "event:" within types as per the JSDoc spec--something I hope can be fixed so it can be surfaced through eslint-plugin-jsdoc (or make a fork if necessary if jsdoctypeparser is not being maintained).

And, FWIW, I've filed an issue at https://github.com/AtomLinter/linter-eslint/issues/1192 for errors reported specifically for eslint-plugin/jsdoc rules--not sure if this is actually your issue or not.

**Comment by gajus** [comment]
Sounds like a reasonable request.

PR welcome.
